packages/api/src/errors.js

```diff
@@ -0,0 +1,102 @@
+import { ErrorCode as MagicErrors } from '@magic-sdk/admin'
+import { JSONResponse } from './utils/json-response.js'
+
+export class HTTPError extends Error {
+  /**
+   *
+   * @param {string} message
+   * @param {number} [status]
+   */
+  constructor (message, status = 500) {
+    super(message)
+    this.name = 'HTTPError'
+    this.status = status
+  }
+
+  /**
+   * @param {Error & {status?: number;code?: string;}} err
+   * @param {import('./env').Env} env
+   */
+  static respond (err, { sentry }) {
+    let error = {
+      code: err.code,
+      message: err.message
+    }
+    let status = err.status || 500
+
+    switch (err.code) {
+      case ErrorUserNotFound.CODE:
+      case ErrorTokenNotFound.CODE:
+        break
+
+      // Magic SDK errors
+      case MagicErrors.TokenExpired:
+        status = 401
+        error.message = 'API Key has expired.'
+        break
+      case MagicErrors.ExpectedBearerString:
+        status = 401
+        error.message =
+          'API Key is missing, make sure the `Authorization` header has a value in the following format `Bearer {api key}`.'
+        break
+      case MagicErrors.MalformedTokenError:
+        status = 401
+        error.message = 'API Key is malformed or failed to parse.'
+        break
+      case MagicErrors.TokenCannotBeUsedYet:
+      case MagicErrors.IncorrectSignerAddress:
+      case MagicErrors.FailedRecoveryProof:
+      case MagicErrors.ApiKeyMissing:
+        status = 401
+        error.code = 'AUTH_ERROR'
+        error.message = 'Authentication failed.'
+        sentry.captureException(err)
+        break
+      case MagicErrors.ServiceError:
+        status = 500
+        error.code = 'SERVER_ERROR'
+        sentry.captureException(err)
+        break
+      default:
+        // catch all server errors
+        if (status >= 500) {
+          error = {
+            code: err.name,
+            message: err.message || 'Server Error'
+          }
+          sentry.captureException(err)
+        } else {
+          // Custom HTTPError
+          error = {
+            code: err.code || 'HTTP_ERROR',
+            message: err.message || 'Server Error'
+          }
+        }
+        break
+    }
+
+    return new JSONResponse({ error }, { status })
+  }
+}
+
+export class ErrorUserNotFound extends Error {
```

Comment:
I think it's much more common to see `Error` as a suffix than a prefix. Including within JS itself (`TypeError`, `RangeError` etc.)

```suggestion
export class UserNotFoundError extends Error {
```